src/packet/secret_key.js

```diff
@@ -199,6 +199,7 @@ SecretKey.prototype.encrypt = function (passphrase) {
   arr.push(crypto.cfb.normalEncrypt(symmetric, key, cleartext, iv));
 
   this.encrypted = util.concatUint8Array(arr);
+  this.isDecrypted = false;
 };
```

Comment:
@tanx i may be wrong but i've noticed that the encrypt() function is not marking the isDecrypted so that obviously my tests with the additions you suggested were not able to work.

I think i've rightly fixd it in this commit. Could you please recheck?